include/field_data.hxx

```diff
@@ -81,6 +82,7 @@ public:
   FieldGeneratorPtr getBndryGenerator(BndryLoc location);
 
 protected:
+  Mesh* fielddatamesh;
```

Comment:
This seems to be a separate addition than the main purpose of this PR (related to #1319 which I think was being delayed).

Reply:
That's right, but is related to this PR in the sense that it's necessary in order to get rid of uses of the global mesh pointer. #1319 has been closed; @ZedThree, @bendudson discussed at the workshop a better re-design of the boundary conditions following the design of the new derivatives. This change is tangential to the redesign of the boundary conditions too; actually I think it may belong more naturally here. The 'correct' final solution may be something like merging `FieldData` into `Field` so we don't have the multiple-inheritance, but as far as I know nobody has immediate plans for doing that...

Reply:
Yes it's coming back to me now -- although the primary goal of this PR isn't to remove uses of the global mesh, just to move the global mesh/dump into a namespace?

Reply:
In my mind the aim was to do both, because in making the namespace all the places where the global mesh was used get flagged up, so I thought it made sense to 'fix' them as I was going through, at least for 'small' fixes.

Adding a `Mesh*` member and a new method to `FieldData` isn't really a 'small' change. Would be happy to back out of it, leave the global mesh pointer for now and fix another time (possibly by merging `Field`/`FieldData`). That would make sense, we'd have two 'design' issues left after the PR, this and the `Solver`, to finally get rid of global mesh pointer from the library. What would people prefer: 1 keep this change as a stop-gap; 2 revert the changes to `FieldData` and fix properly in a redesign (addressing #897); 3 something else?

Reply:
My preference would be to back out on this for now (we don't support no global mesh until everywhere supports it anyway, so I guess this isn't really blocking that) and address it in a redesign alongside the updates to boundary (as this may impact on how we want to do this anyway).

Reply:
I've taken out the changes to FieldData. Now it just uses `bout::globals::mesh` to use the global mesh pointer.